clients/python/src/osparc/_utils.py

```diff
@@ -46,38 +48,40 @@ def __init__(
         self._client: httpx.Client = httpx.Client(
             auth=auth, base_url=base_url, follow_redirects=True
         )
+        self._page: Optional[Page] = None
+        self._page_item_counter = 0
 
     def __del__(self):
         self._client.close()
 
+    def __next__(self):
+        if self._page is None:
+            self._page = self._first_page_callback()
+        if self._page_item_counter > (len(self._page.items) - 1):
+            next_page_url = self._page.links.next
+            if next_page_url is None:
+                self._page = None
+                self._page_item_counter = 0
+                raise StopIteration
+            response = self._client.get(next_page_url)
+            self._page = self._api_client._ApiClient__deserialize(
+                response.json(), type(self._page)
+            )
+            self._page_item_counter = 0
+        next_item = self._page.items[self._page_item_counter]
+        self._page_item_counter += 1
+        return next_item
```

Comment:
I am curious. Perhaps you can explain me why you moved from `__iter__` to `__next__` offline?

Reply:
Because you suggested that I do 😉. It was sparked by this comment: https://github.com/ITISFoundation/osparc-simcore-clients/pull/216#discussion_r1844021191. Now the `PaginationIterator` is a proper iterator, meaning I can call `next` on it and it inherits from `collections.abc.Iterator`. Before I couldn't call `next` on it, I had to always loop over it, even if `len(pagination_iterator)=1`. So this should be a little more userfriendly